pkg/streamingpromql/operators/aggregation_test.go

```diff
@@ -174,31 +176,35 @@ func TestAggregation_GroupLabelling(t *testing.T) {
 			inputSeries:          labels.FromStrings(labels.MetricName, "my_metric", "env", "prod", "a-label", "a-value", "d-label", "d-value", "f-label", "f-value"),
 			expectedOutputSeries: labels.FromStrings("a-label", "a-value", "d-label", "d-value", "f-label", "f-value"),
 		},
-		"grouping with 'without', multiple grouping labels, input has all grouping labels": {
+		"grouping with 'without', multiple grouping labels, input has superset of grouping labels": {
 			grouping:             []string{"cluster", "env"},
 			without:              true,
 			inputSeries:          labels.FromStrings(labels.MetricName, "my_metric", "env", "prod", "cluster", "cluster-1", "a-label", "a-value", "d-label", "d-value", "f-label", "f-value"),
 			expectedOutputSeries: labels.FromStrings("a-label", "a-value", "d-label", "d-value", "f-label", "f-value"),
 		},
+		"grouping with 'without', multiple grouping labels, input has all grouping labels and no others": {
+			grouping:             []string{"cluster", "env"},
+			without:              true,
+			inputSeries:          labels.FromStrings(labels.MetricName, "my_metric", "env", "prod", "cluster", "cluster-1"),
+			expectedOutputSeries: labels.EmptyLabels(),
+		},
 	}
 
 	for name, testCase := range testCases {
 		t.Run(name, func(t *testing.T) {
-			require.True(t, slices.IsSorted(testCase.grouping), "invalid test case: grouping labels must be sorted")
-
-			aggregator := &Aggregation{
-				Grouping: testCase.grouping,
-				Without:  testCase.without,
-			}
-
-			labelsFunc := aggregator.seriesToGroupLabelsFunc()
-			stringFunc := aggregator.seriesToGroupLabelsStringFunc()
+			aggregator := NewAggregation(nil, time.Time{}, time.Time{}, time.Minute, testCase.grouping, testCase.without, nil)
```

Comment:
Now that we're using `NewAggregation` we should use some test cases where the labels are not sorted.

Reply:
Added in fff95c4